pkg/cmd/playback.go

```diff
@@ -0,0 +1,196 @@
+	pc.cmd.Flags().StringVar(&pc.webhookAddress, "forward-to", fmt.Sprintf("localhost:%d", defaultWebhookPort), "Address to forward webhooks to")
+	pc.cmd.Flags().StringVar(&pc.filepath, "cassette", "default_cassette.yaml", "The cassette file to use")
+	pc.cmd.Flags().StringVar(&pc.cassetteDir, "cassette-root-dir", "", "Directory to store all cassettes in. Relative cassette paths are considered relative to this directory.")
+
+	// // Hidden configuration flags, useful for dev/debugging
+	pc.cmd.Flags().StringVar(&pc.apiBaseURL, "api-base", "https://api.stripe.com", "Sets the API base URL")
+
+	return pc
+}
+
+func (pc *playbackCmd) runPlaybackCmd(cmd *cobra.Command, args []string) error {
+	fmt.Println()
+	fmt.Println("Seting up playback server...")
+	fmt.Println()
+
+	// --- Validate command-line args
+	// Check that mode is valid
+	if pc.mode != playback.Auto && pc.mode != playback.Record && pc.mode != playback.Replay {
+		return fmt.Errorf(
+			"\"%v\" is not a valid mode. It must be either \"%v\", \"%v\", or \"%v\"",
+			pc.mode, playback.Auto, playback.Record, playback.Replay)
+	}
+
+	// Check that cassette root directory is valid
+	absoluteCassetteDir, err := filepath.Abs(pc.cassetteDir)
+	if err != nil {
+		return fmt.Errorf("Error with --cassette-root-dir: %w", err)
+	}
+
+	handle, err := os.Stat(absoluteCassetteDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return fmt.Errorf("the directory \"%v\" does not exist. Please create it, then re-run the command", absoluteCassetteDir)
+		}
+		return fmt.Errorf("Unexpected error when checking --cassette-root-dir: %w", err)
+	}
+
+	if !handle.Mode().IsDir() {
+		return fmt.Errorf("The provided `--cassette-root-dir` option is not a valid directory: %v", absoluteCassetteDir)
+	}
+
+	// --- Start up the playback HTTP server
+	// TODO: `playback` should handle setup (and teardown) of `stripe listen` as well
+	addressString := pc.address
+	remoteURL := pc.apiBaseURL
+
+	webhookAddress := "http://" + pc.webhookAddress
+
+	httpWrapper, err := playback.NewServer(remoteURL, webhookAddress, absoluteCassetteDir)
+	if err != nil {
+		return nil
+	}
+
+	server := httpWrapper.InitializeServer(addressString)
+	go func() {
+		err = server.ListenAndServe()
+		fmt.Fprint(os.Stderr, err.Error())
+		os.Exit(1)
+	}()
+
+	fullAddressString := "http://" + addressString
+
+	client := &http.Client{}
+
+	// --- Use the flag values to configure the server (over HTTP)
+	resp, err := client.Get(fullAddressString + "/playback/mode/" + pc.mode)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return errors.New("Non 200 status code received during startup: " + resp.Status)
+	}
+
+	resp, err = client.Get(fullAddressString + "/playback/cassette/load?filepath=" + pc.filepath)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return errors.New("Non 200 status code received during startup: " + resp.Status)
+	}
```

Comment:
(I think we lost the comment you put here earlier Richard, but it still unresolved so paraphrasing again): 

We should change this so that we can pass these values to the playback.Server on initialization too, and not have to do HTTP calls when starting up the server for the first time.

Reply:
Also, I think there are race conditions where the server (which starts up in the goroutine above) doesn't start listening before we send these startup HTTP requests - so a even better reason to get rid of these...

Reply:
Fixed in commit 63e526b5aebc8a7a7fc2122a50090d172df49aa9.